src/Notecard.cpp

```diff
@@ -404,72 +383,126 @@ J *Notecard::requestAndResponseWithRetry(J *req, uint32_t timeoutSeconds)
 {
-    NoteDebug(message);
+    return NoteRequest(req);
 }
 
 /**************************************************************************/
 /*!
-    @brief  Write a formatted message to the serial debug stream.
-    @param    format
-              A format string to log to the serial debug stream.
-    @param    ... one or more values to interpolate into the format string.
+    @brief  Sends a request to the Notecard, retrying it on failure until the
+            provided timeout interval lapses.
+    @param    req
+              A `J` JSON request object.
+    @param    timeoutSeconds
+              The timeout interval, in seconds.
+    @return `True` if the message was successfully sent to the Notecard,
+            `False` if the message couldn't be sent.
 */
 /**************************************************************************/
-void Notecard::logDebugf(const char *format, ...)
+bool Notecard::sendRequestWithRetry(J *req, uint32_t timeoutSeconds)
 {
-    char message[256];
-    va_list args;
-    va_start(args, format);
-    vsnprintf(message, sizeof(message), format, args);
-    va_end(args);
-    NoteDebug(message);
+    return NoteRequestWithRetry(req, timeoutSeconds);
 }
 
 /**************************************************************************/
 /*!
-    @brief  Periodically show Notecard sync status, returning `TRUE`
-            if something was displayed to the debug stream.
-    @param    pollFrequencyMs
-              The frequency to poll the Notecard for sync status.
-    @param    maxLevel
-              The maximum log level to output to the debug console. Pass
-              -1 for all.
-    @return `True` if a pending response was displayed to the debug stream.
+    @brief  Set the debug output source.
+            A NoteLog object will be constructed via `make_note_log()`
+            using a platform specific logging channel (for example, `Serial`
+            on Arduino). The specified channel will be configured as the
+            source for debug messages provided to `notecard.logDebug()`.
+    @param    noteLog
+              A platform specific log implementation to be used for
+              debug output.
 */
 /**************************************************************************/
-bool Notecard::debugSyncStatus(int pollFrequencyMs, int maxLevel)
+void Notecard::setDebugOutputStream(NoteLog * noteLog_)
 {
-    return NoteDebugSyncStatus(pollFrequencyMs, maxLevel);
+    noteLog = noteLog_;
+    if (noteLog) {
+        NoteSetFnDebugOutput(noteLogPrint);
+    } else {
+        NoteSetFnDebugOutput(nullptr);
+    }
 }
 
 /**************************************************************************/
 /*!
-    @brief  Determines if there is an error string present in a response object.
-    @param    rsp
-              A `J` JSON Response object.
-    @return `True` if the response object contains an error.
+    @brief  Set the lock/unlock functions the Notecard uses for I2C access.
+    @param    lockI2cFn
```

Comment:
Should this read lockI2cFn_ (with the trailing underscore)? Also, is the trailing underscore your way of indicating that the variable is a function parameter?

Reply:
> Should this read lockI2cFn_ (with the trailing underscore)?

That would be to taste. I don't put the trailing underscore in the `.h` file, which is typically where I put Doxygen comments.

> Also, is the trailing underscore your way of indicating that the variable is a function parameter?

Yes.